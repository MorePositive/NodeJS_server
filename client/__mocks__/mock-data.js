export const mockItem = [{
  firstName: "Ivan",
  lastName: "Ivanov",
  phone: 50887562,
  email: "asd@ads.com"
}];

export const mockItems = [{
  firstName: "Oleg",
  lastName: "Olegov",
  phone: 68333121,
  email: "mail@test.com"
},
{
  firstName: "John",
  lastName: "Smith",
  phone: 67125748,
  email: "test@mail.com"
}];

export const mockTitle = "sunt aut facere repellat provident occaecati excepturi optio reprehenderit";

export const mockBody = "quia et suscipit suscipit recusandae consequuntur expedita et cum reprehenderit molestiae ut ut quas totam nostrum rerum est autem sunt rem eveniet architecto";
