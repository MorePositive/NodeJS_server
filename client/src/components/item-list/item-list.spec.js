import React from 'react';

import ItemList from './item-list';
import { mockItem, mockItems } from '../../../__mocks__/mock-data'

describe("ItemList component", () => {
  let component;

  beforeEach(() => {
    component = shallow(<ItemList data={[]} />);
  });

  it("should render ItemList component", () => {
    const wrapper = component.find(".item-list");
    expect(wrapper).toHaveLength(1);
  });

  it("should contain at least 1 paragraph", () => {
    const wrapper = component.find("p");
    expect(wrapper.length).toBeGreaterThanOrEqual(1);
  });

  it("should render no-data paragraph", () => {
    const wrapper = component.find(".no-data");
    expect(wrapper).toMatchSnapshot();
  });
  
  it("should render item", () => {
    component = shallow(<ItemList data={mockItem} />)
    const wrapper = component.find(".card");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 2 items", () => {
    component = shallow(<ItemList data={mockItems} />);
    const item = component.find(".card");
    expect(item.length).toBe(2);
  });
});
