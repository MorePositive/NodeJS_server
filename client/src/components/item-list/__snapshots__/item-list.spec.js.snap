// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList component should render item 1`] = `
<ul
  className="card"
  key="asd@ads.com"
>
  <li
    className="card-item"
  >
    Ivan
  </li>
  <li
    className="card-item"
  >
    Ivanov
  </li>
  <li
    className="card-item"
  >
    50887562
  </li>
  <li
    className="card-item"
  >
    asd@ads.com
  </li>
</ul>
`;

exports[`ItemList component should render no-data paragraph 1`] = `
<p
  className="no-data"
>
  There is no data yet
</p>
`;
